src/test/web/customReporter.ts

```diff
@@ -0,0 +1,178 @@
+    | {
+          event: typeof constants.EVENT_TEST_FAIL;
+          title: string;
+          err: Exception;
+          duration?: number;
+          titlePath: string[];
+          fullTitle: string;
+          slow: number;
+      }
+    | {
+          event: typeof constants.EVENT_TEST_PENDING;
+          title: string;
+          titlePath: string[];
+          fullTitle: string;
+          slow: number;
+      }
+    | {
+          event: typeof constants.EVENT_TEST_PASS;
+          title: string;
+          duration?: number;
+          titlePath: string[];
+          fullTitle: string;
+          slow: number;
+      };
+let currentPromise = Promise.resolve();
+const messages: Message[] = [];
+let url = '';
+
+function writeReportProgress(message: Message) {
+    if (env.uiKind === UIKind.Desktop) {
+        messages.push(message);
+        if (message.event === constants.EVENT_RUN_END) {
+            const ext = extensions.getExtension(JVSC_EXTENSION_ID_FOR_TESTS)!.extensionUri;
+            const logFile = Uri.joinPath(ext, 'webtest.json');
+            workspace.fs.writeFile(logFile, Buffer.from(JSON.stringify(messages))).then(noop, noop);
+        }
+    } else {
+        if (message.event === constants.EVENT_RUN_BEGIN) {
+            console.log(
+                `DEBUG_JUPYTER_SERVER_URI={workspace.getConfiguration('jupyter').get('DEBUG_JUPYTER_SERVER_URI')}`
+            );
+            const reportServerPor = workspace.getConfiguration('jupyter').get('REPORT_SERVER_PORT') as number;
+
+            url = `http://127.0.0.1:${reportServerPor}`;
+            console.error(`Started test reporter and writing to ${url}`);
+        }
+        currentPromise = currentPromise.finally(() => {
+            return fetch(url, {
+                method: 'post',
+                body: JSON.stringify(message),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }).catch((_ex) => {
+                // console.error(`Failed to post data to ${url}`, ex);
+            });
+        });
+    }
+}
+function formatException(err: any) {
+    const props = ['actual', 'expected', 'operator', 'generatedMessage'];
+    let message = '';
+    if (typeof err.inspect === 'function') {
+        message = err.inspect() + '';
+    } else if (err.message && typeof err.message.toString === 'function') {
+        message = err.message + '';
+    }
+    const error: Record<string, any> = {
+        name: err.name,
+        message,
+        stack: err.stack,
+        showDiff: err.showDiff,
+        inspect: ''
+    };
+    props.forEach((prop) => {
+        try {
+            error[prop] = JSON.parse(JSON.stringify(err[prop]));
+        } catch {
+            error[prop] = '<..?..>';
+        }
+    });
+    return error as Exception;
+}
+
+function CustomReporter(this: any, runner: mochaTypes.Runner, options: mochaTypes.MochaOptions) {
```

Comment:
Can use ES6 classes, as we cannot import the mocha reporter writers in web, have to use the good old fashioned `require` to import and then extend (again using mocha dependencies to inherit instead of pulling any node specific modules to inherit stuff - as this needs to work in the browser as well).